### ISSUE TITLE
Some fields are dropped when exporting to Hayagriva

### ISSUE THREAD

**Issue by Enivex** [JabRef/jabref#10596]
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

23H2

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Create a .bib file containing the following
```
@Article{Euler1757,
  author  = {Euler, Leonhard},
  journal = {Mémoires de l'académie des sciences de Berlin},
  title   = {Principes généraux du mouvement des fluides},
  year    = {1757},
  pages   = {274--315},
}
```
2. Export to Hayagriva yaml
3. The result is
```
---
Euler1757:
  type: article
  title: "Principes généraux du mouvement des fluides"
  author:
    - Euler, Leonhard
  date: 1757
  page-range: 274--315
---
```

As is evident, the journal is not there. In comparison, the output from the hayagriva CLI is
```
---
Euler1757:
  type: article
  author: "Euler, Leonhard"
  parent:
    - type: periodical
      title:
        value: "Mémoires de l'académie des sciences de Berlin"
        sentence-case: "Mémoires de l'académie des sciences de berlin"
  date: 1757
  title: Principes généraux du mouvement des fluides
  page-range: 274-315
```

JabRef 5.12--2023-10-26--d6a0e59
Windows 11 10.0 amd64 
Java 21.0.1 
JavaFX 20.0.2+3

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

**Comment by Siedlerchr** [comment]
Seems like the field journal is missing here https://github.com/JabRef/jabref/blob/b6d8f826b71b6489b636b01e3da8c26833ccb117/src/main/resources/resource/layout/hayagrivayaml.layout 

Needs to be added and probably the tests need to be adjusted as well

**Comment by koppor** [comment]
@JawadTUE Are you still looking for an issue? Maybe, this is a good follow-up.

**Comment by JawadTUE** [comment]
Yes, I implemented the Hayagriva yaml in #10451, but I left the `parent` field out. Seems like I should have added that anyway. So please assign me to this issue so I can try to fix it.
